### PR TITLE
LPD-36166 Relations added between Objects and SystemObjects fail OpenAPI YAML Validation

### PR DESCRIPTION
**Pull request by dannielraposo**
**What's changed?:**

- The generated OpenAPIs for system objects and custom object relationships now correctly include required fields such as the `schema` and `description` attributes in the path parameters and response sections, ensuring proper validation and compliance with OpenAPI standards.
- The logic now dynamically retrieves the data type from the system object’s basic endpoints (e.g., `getUserAccount` for User) to ensure consistency. This resolves the issue where _id_ fields, although `long` in the database, were marked by default as `string` in the OpenAPI specification. In case of future updates to the `RestBuilder` (including proper swagger annotations), this approach ensures automatic alignment with the parameter type.
- A test has been added to validate that the OpenAPI is generated correctly, ensuring all necessary fields, such as `description` and `schema`, are present in the dynamic endpoints.

**How to test it?:**

- The behavior can be tested by creating a custom object, establishing a relationship with a system object (e.g., User), and verifying the OpenAPI output to confirm that the `schema` and `description` fields are properly defined. The OpenAPI can also be validated with the [Swagger & OpenAPI Online Validator](https://apitools.dev/swagger-parser/online)
- The added test ensures this behavior is consistent and correct.

See the following **Jira Ticket**: [LPD-36166](https://liferay.atlassian.net/jira/software/c/projects/LPD/issues/LPD-36166)

### CODE REVIEW

**Comment by dannielraposo** [comment]
ci:test:relevant

**Comment by dannielraposo** [comment]
ci:test:relevant

**Comment by dannielraposo** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/OpenAPIResourceTest.java [hunk 1]
```diff
@@ -281,6 +329,18 @@ private void _assertOpenAPI(
 			JSONCompareMode.STRICT);
 	}
 
+	private void _assertOpenAPI(String fileName, String endpoint)
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/OpenAPIResourceTest.java:332]: You are invoking this method only once, and you are calling just one method, better to get rid of it

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/RelatedObjectEntryOpenAPIContributor.java [hunk 2]
```diff
@@ -145,14 +146,45 @@ private void _contribute(
...
+						_getIdParameterName(schemaName), parameter.getName())) {
+
+					parameterSchema = parameter.getSchema();
+
+					break;
+				}
+			}
+		}
+
+		if (parameterSchema == null) {
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/RelatedObjectEntryOpenAPIContributor.java:172]: There is a class called StringSchema that you may use

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/RelatedObjectEntryOpenAPIContributor.java [hunk 3]
```diff
@@ -145,14 +146,45 @@ private void _contribute(
 			_getIdParameterTemplate(schemaName), StringPool.SLASH,
 			systemObjectRelationship.getName());
 
+		Schema parameterSchema = null;
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/RelatedObjectEntryOpenAPIContributor.java:149]: Maybe better to use a more descriptive name, for example idSchema ? (I am not sure, but I think Brian will like more) xD

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/RelatedObjectEntryOpenAPIContributor.java [hunk 4]
```diff
@@ -145,14 +146,45 @@ private void _contribute(
...
+
+		PathItem pathItem = paths.get(
+			StringBundler.concat(
+				StringPool.SLASH, version, StringPool.SLASH,
+				jaxRsApplicationDescriptor.getPath(), StringPool.SLASH,
+				_getIdParameterTemplate(schemaName)));
+
+		Operation getOperation = (pathItem != null) ? pathItem.getGet() : null;
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/RelatedObjectEntryOpenAPIContributor.java:157]: I think we usually don't use ternary operators. Please, have a look at the portal and it we don't use it, please refactor

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/RelatedObjectEntryOpenAPIContributor.java [hunk 5]
```diff
@@ -145,14 +146,45 @@ private void _contribute(
...
+
+		if ((getOperation != null) && (getOperation.getParameters() != null)) {
+			for (Parameter parameter : getOperation.getParameters()) {
+				if ((parameter != null) &&
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/RelatedObjectEntryOpenAPIContributor.java:161]: I am not sure if parameter could be null

**Comment by ccorreagg** [comment]
I am executing the test and it is failing:

```gw tI --tests OpenAPIResourceTest.testGetOpenAPIWithSystemObjectRelationship```

```
com.liferay.object.rest.internal.resource.v1_0.test.OpenAPIResourceTest > testGetOpenAPIWithSystemObjectRelationship STARTED

com.liferay.object.rest.internal.resource.v1_0.test.OpenAPIResourceTest > testGetOpenAPIWithSystemObjectRelationship FAILED
    java.lang.AssertionError: Thread Thread[http-nio-8080-exec-4,5,main] caught concurrent failure: java.lang.AssertionError: {level=ERROR, loggerName=com.liferay.portal.vulcan.internal.jaxrs.exception.mapper.ExceptionMapper, message=java.lang.String cannot be cast to io.swagger.v3.oas.models.OpenAPI
    java.lang.AssertionError: {level=ERROR, loggerName=com.liferay.portal.vulcan.internal.jaxrs.exception.mapper.ExceptionMapper, message=java.lang.String cannot be cast to io.swagger.v3.oas.models.OpenAPI
    	at com.liferay.portal.test.rule.LogAssertionTestRule$LogAppender.append(LogAssertionTestRule.java:318)
    	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:161)
    	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:134)
    	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:125)
    	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:89)
    	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:542)
    	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:500)
    	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:483)
    	at org.apache.logging.log4j.core.config.LoggerConfig.logParent(LoggerConfig.java:533)
    	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:502)
    	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:483)
    	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:417)
    	at org.apache.logging.log4j.core.config.AwaitCompletionReliabilityStrategy.log(AwaitCompletionReliabilityStrategy.java:82)
    	at org.apache.logging.log4j.core.Logger.log(Logger.java:161)
    	at org.apache.logging.log4j.spi.AbstractLogger.tryLogMessage(AbstractLogger.java:2205)
    	at org.apache.logging.log4j.spi.AbstractLogger.logMessageTrackRecursion(AbstractLogger.java:2159)
    	at org.apache.logging.log4j.spi.AbstractLogger.logMessageSafely(AbstractLogger.java:2142)
    	at org.apache.logging.log4j.spi.AbstractLogger.logMessage(AbstractLogger.java:2017)
    	at org.apache.logging.log4j.spi.AbstractLogger.logIfEnabled(AbstractLogger.java:1983)
    	at com.liferay.portal.log.Log4jLogImpl.error(Log4jLogImpl.java:56)
    	at com.liferay.portal.kernel.log.LogWrapper.error(LogWrapper.java:70)
    	at com.liferay.portal.log.Log4jLogContextLogWrapper.error(Log4jLogContextLogWrapper.java:83)
    	at com.liferay.portal.kernel.log.LogWrapper.error(LogWrapper.java:70)
    	at com.liferay.portal.kernel.log.SanitizerLogWrapper.error(SanitizerLogWrapper.java:110)
    	at com.liferay.portal.vulcan.internal.jaxrs.exception.mapper.ExceptionMapper.getProblem(ExceptionMapper.java:59)
    	at com.liferay.portal.vulcan.internal.jaxrs.exception.mapper.ExceptionMapper.getProblem(ExceptionMapper.java:20)
    	at com.liferay.portal.vulcan.jaxrs.exception.mapper.BaseExceptionMapper.toResponse(BaseExceptionMapper.java:26)
    	at com.liferay.portal.vulcan.internal.jaxrs.exception.mapper.ExceptionMapper.toResponse(ExceptionMapper.java:43)
    	at com.liferay.portal.vulcan.internal.jaxrs.exception.mapper.ExceptionMapper.toResponse(ExceptionMapper.java:20)
    	at com.liferay.portal.vulcan.internal.jaxrs.exception.mapper.ExceptionMapper.toResponse(ExceptionMapper.java:51)
    	at com.liferay.portal.vulcan.internal.jaxrs.exception.mapper.ExceptionMapper.toResponse(ExceptionMapper.java:20)
    	at org.apache.cxf.jaxrs.utils.ExceptionUtils.convertFaultToResponse(ExceptionUtils.java:85)
    	at org.apache.cxf.jaxrs.utils.JAXRSUtils.convertFaultToResponse(JAXRSUtils.java:1878)
    	at org.apache.cxf.jaxrs.JAXRSInvoker.handleFault(JAXRSInvoker.java:371)
    	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:218)
    	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:104)
    	at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59)
    	at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96)
    	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:307)
    	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121)
    	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:265)
    	at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234)
    	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208)
    	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160)
    	at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:225)
    	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:304)
    	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doGet(AbstractHTTPServlet.java:222)
    	at javax.servlet.http.HttpServlet.service(HttpServlet.java:529)
    	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:279)
    	at org.eclipse.equinox.http.servlet.internal.registration.EndpointRegistration.service(EndpointRegistration.java:147)
    	at org.eclipse.equinox.http.servlet.internal.servlet.FilterChainImpl.doFilter(FilterChainImpl.java:50)
    	at com.liferay.portal.security.auth.verifier.internal.tracker.AuthVerifierFilterTracker$RemoteAccessFilter.doFilter(AuthVerifierFilterTracker.java:213)
    	at org.eclipse.equinox.http.servlet.internal.registration.FilterRegistration.doFilter(FilterRegistration.java:120)
    	at org.eclipse.equinox.http.servlet.internal.servlet.FilterChainImpl.doFilter(FilterChainImpl.java:45)
    	at com.liferay.portal.security.auth.verifier.internal.tracker.AuthVerifierFilterTracker$AuditFilter.doFilter(AuthVerifierFilterTracker.java:182)
    	at org.eclipse.equinox.http.servlet.internal.registration.FilterRegistration.doFilter(FilterRegistration.java:120)
    	at org.eclipse.equinox.http.servlet.internal.servlet.FilterChainImpl.doFilter(FilterChainImpl.java:45)
    	at com.liferay.portal.kernel.servlet.BaseFilter.processFilter(BaseFilter.java:152)
    	at com.liferay.portal.servlet.filters.authverifier.AuthVerifierFilter.processFilter(AuthVerifierFilter.java:203)
    	at com.liferay.portal.kernel.servlet.BaseFilter.doFilter(BaseFilter.java:42)
    	at org.eclipse.equinox.http.servlet.internal.registration.FilterRegistration.doFilter(FilterRegistration.java:120)
    	at org.eclipse.equinox.http.servlet.internal.servlet.FilterChainImpl.doFilter(FilterChainImpl.java:45)
    	at org.eclipse.equinox.http.servlet.internal.servlet.ResponseStateHandler.processRequest(ResponseStateHandler.java:71)
    	at org.eclipse.equinox.http.servlet.internal.context.DispatchTargets.doDispatch(DispatchTargets.java:120)
    	at com.liferay.portal.osgi.web.http.servlet.internal.servlet.HttpServletEndpointServlet.service(HttpServletEndpointServlet.java:64)
    	at javax.servlet.http.HttpServlet.service(HttpServlet.java:623)
    	at com.liferay.portal.module.framework.ModuleFrameworkServletAdapter.service(ModuleFrameworkServletAdapter.java:45)
    	at javax.servlet.http.HttpServlet.service(HttpServlet.java:623)
    	at com.liferay.shielded.container.internal.proxy.ServletWrapper.service(ServletWrapper.java:103)
    	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:199)
    	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
    	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
    	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:168)
    	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:117)
    	at com.liferay.portal.kernel.servlet.BaseFilter.processFilter(BaseFilter.java:152)
    	at com.liferay.change.tracking.internal.servlet.filter.CTCollectionPreviewFilter.processFilter(CTCollectionPreviewFilter.java:57)
    	at com.liferay.portal.kernel.servlet.BaseFilter.doFilter(BaseFilter.java:42)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDoFilter(InvokerFilterChain.java:204)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:109)
    	at com.liferay.portal.remote.jaxrs.whiteboard.internal.servlet.filter.JAXRSActivationFilter.doFilter(JAXRSActivationFilter.java:73)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDoFilter(InvokerFilterChain.java:204)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:109)
    	at com.liferay.portal.servlet.filters.password.modified.PasswordModifiedFilter.processFilter(PasswordModifiedFilter.java:47)
    	at com.liferay.portal.kernel.servlet.BaseFilter.doFilter(BaseFilter.java:42)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDoFilter(InvokerFilterChain.java:204)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:109)
    	at com.liferay.portal.servlet.filters.lockout.LockoutFilter.processFilter(LockoutFilter.java:49)
    	at com.liferay.portal.kernel.servlet.BaseFilter.doFilter(BaseFilter.java:42)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDoFilter(InvokerFilterChain.java:204)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:109)
    	at com.liferay.portal.kernel.servlet.BaseFilter.processFilter(BaseFilter.java:152)
    	at com.liferay.portal.servlet.filters.virtualhost.VirtualHostFilter.processFilter(VirtualHostFilter.java:243)
    	at com.liferay.portal.kernel.servlet.BaseFilter.doFilter(BaseFilter.java:42)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDoFilter(InvokerFilterChain.java:204)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:109)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDirectCallFilter(InvokerFilterChain.java:189)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:92)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDirectCallFilter(InvokerFilterChain.java:189)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:92)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDirectCallFilter(InvokerFilterChain.java:168)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:92)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDirectCallFilter(InvokerFilterChain.java:189)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:92)
    	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilter.doFilter(InvokerFilter.java:114)
    	at com.liferay.shielded.container.internal.proxy.FilterWrapper.doFilter(FilterWrapper.java:69)
    	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:168)
    	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
    	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168)
    	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
    	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:481)
    	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
    	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
    	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
    	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:346)
    	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:388)
    	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
    	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:936)
    	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1791)
    	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
    	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
    	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
    	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
    	at java.lang.Thread.run(Thread.java:750)
    Caused by: java.lang.ClassCastException: java.lang.String cannot be cast to io.swagger.v3.oas.models.OpenAPI
    	at com.liferay.object.rest.internal.openapi.v1_0.ObjectEntryOpenAPIResourceImpl.getSchemas(ObjectEntryOpenAPIResourceImpl.java:138)
    	at com.liferay.object.rest.internal.vulcan.openapi.contributor.RelatedObjectEntryOpenAPIContributor._contribute(RelatedObjectEntryOpenAPIContributor.java:134)
    	at com.liferay.object.rest.internal.vulcan.openapi.contributor.RelatedObjectEntryOpenAPIContributor.contribute(RelatedObjectEntryOpenAPIContributor.java:99)
    	at com.liferay.portal.vulcan.internal.resource.OpenAPIResourceImpl._getOpenAPI(OpenAPIResourceImpl.java:764)
    	at com.liferay.portal.vulcan.internal.resource.OpenAPIResourceImpl.getOpenAPI(OpenAPIResourceImpl.java:120)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:498)
    	at com.liferay.headless.admin.user.internal.resource.v1_0.OpenAPIResourceImpl.getOpenAPI(OpenAPIResourceImpl.java:65)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:498)
    	at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179)
    	at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96)
    	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:201)
    	... 89 more

    	at org.junit.Assert.fail(Assert.java:89)
    	at com.liferay.portal.test.rule.LogAssertionTestRule.endAssert(LogAssertionTestRule.java:119)
    	at com.liferay.portal.test.rule.LogAssertionTestRule.afterClass(LogAssertionTestRule.java:176)
    	at com.liferay.portal.test.rule.LogAssertionTestRule.afterMethod(LogAssertionTestRule.java:183)
    	at com.liferay.portal.test.rule.LogAssertionTestRule.afterMethod(LogAssertionTestRule.java:40)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:97)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$1.evaluate(TestExecutorRunnable.java:228)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable._execute(TestExecutorRunnable.java:307)
    	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable.run(TestExecutorRunnable.java:86)
    	at java.lang.Thread.run(Thread.java:750)
    	Suppressed: java.lang.AssertionError: 
    Expected: components
         but none found
     ; 
    Expected: info
         but none found
     ; 
    Expected: openapi
         but none found
     ; 
    Expected: paths
         but none found
     ; 
    Expected: servers
         but none found
     ; 
    Unexpected: status
     ; 
    Unexpected: title

    		at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:222)
    		at com.liferay.object.rest.internal.resource.v1_0.test.OpenAPIResourceTest.testGetOpenAPIWithSystemObjectRelationship(OpenAPIResourceTest.java:305)
    		at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    		at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    		at java.lang.reflect.Method.invoke(Method.java:498)
    		at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.java:344)
    		at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
    		at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    		at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    		at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    		at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    		... 19 more

1 test completed, 1 failed
There were failing tests. See the report at: file:///home/carlos/Liferay/Proyecto/liferay-portal/modules/apps/object/object-rest-test/build/reports/tests/testIntegration/index.html

> Task :apps:object:object-rest-test:stopTestableTomcat
> Task :apps:object:object-rest-test-util:autoUpdateXml SKIPPED
> Task :apps:object:object-test-util:autoUpdateXml SKIPPED
> Task :apps:portal-configuration:portal-configuration-test-util:autoUpdateXml SKIPPED
> Task :apps:portal-odata:portal-odata-api:autoUpdateXml SKIPPED
> Task :apps:portal-search:portal-search-api:autoUpdateXml SKIPPED
> Task :apps:portal-vulcan:portal-vulcan-api:autoUpdateXml SKIPPED
> Task :apps:headless:headless-admin-taxonomy:headless-admin-taxonomy-client:autoUpdateXml SKIPPED
> Task :apps:headless:headless-admin-user:headless-admin-user-api:autoUpdateXml SKIPPED


BUILD SUCCESSFUL in 43s
199 actionable tasks: 5 executed, 194 up-to-date
```

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/OpenAPIResourceTest.java [hunk 6]
```diff
@@ -267,6 +273,56 @@ public void testGetOpenAPIWithActions() throws Exception {
...
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY, null);
+
+		JaxRsApplicationDescriptor jaxRsApplicationDescriptor =
+			_userSystemObjectDefinitionManager.getJaxRsApplicationDescriptor();
+
+		String endpoint = StringBundler.concat(
+			jaxRsApplicationDescriptor.getApplicationPath(), StringPool.SLASH,
+			jaxRsApplicationDescriptor.getVersion(), "/openapi.json");
+
+		JSONAssert.assertEquals(
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/OpenAPIResourceTest.java:305]: I think it is not a good idea to include a strict validation here, as if other team modify the UserAccount schema or includes a new endpoint, our test will fail.

I think the best choice is to get rid of all the fields that do not depend on us, and just validating the fields that applies to the System Object case. That is, for example:

- Object1 schema with all the properties
- PageObject1 schema with all the properties
- UserAccount schema with the relationships
- All the relationship endpoints (get, put and delete)

And then, you can modify the type of comparison from STRICT to LENIENT (as the order does not matter either in the JSON file)

image

-

Comment (ccorreagg) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/OpenAPIResourceTest.java:305]: I also miss the cases for the other relationships.

For example:

One UserAccount Many Object1 (`relation1ToM`)
Many UserAccount Many Object1 (`relationMtoM`)

Then, we could rename from `relation1` to `relationMto1` to match the pattern and provide more info of which property belongs to which relationship type

**Review by ccorreagg — CHANGES_REQUESTED** [review]
Please, have a look at my comments related to the test

**Comment by ccorreagg** [comment]
Please, rebase with master and re-execute the tests. I think those errors should be fixed already

**Comment by dannielraposo** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:stop

**Comment by ccorreagg** [comment]
ci:stop

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
image

The failures are not related

**Comment by ccorreagg** [comment]
ci:forward:force

**Pull request closed without merging**